### PR TITLE
[topi] fix strategy for sparse dense cuda

### PR DESCRIPTION
**Pull request by antinucleon**

### CODE REVIEW

**Comment by antinucleon** [comment]
@vinx13 @ceruleangu 

I fixed the strategy so it is able to lower now. However, the schedule is not working with the current master. Could you fix it?

To verify it, please enable CUDA test at:
https://github.com/apache/incubator-tvm/blob/master/tests/python/relay/test_sparse_dense_convert.py#L50

File: python/tvm/relay/op/strategy/generic.py [hunk 1]
```diff
@@ -599,12 +599,30 @@ def batch_matmul_strategy(attrs, inputs, out_type, target):
...
+    """sparse dense generic strategy"""
+    logger.warning("sparse dense is not optimized for this platform.")
+    strategy = _op.OpStrategy()
+    strategy.add_implementation(wrap_compute_sparse_dense(topi.nn.sparse_dense),
+                                wrap_topi_schedule(topi.generic.schedule_sparse_dense),
+                                name="sparse_dense.generic")
+    return strategy
+
+
+# # sparse_dense
```
Comment (junrushao1994) [inline python/tvm/relay/op/strategy/generic.py:620]: remove those lines?

**Comment by tqchen** [comment]
@antinucleon please updater as per comment

**Comment by antinucleon** [comment]
Updated

File: python/tvm/relay/op/strategy/x86.py [hunk 2]
```diff
@@ -294,11 +294,16 @@ def batch_matmul_strategy_cpu(attrs, inputs, out_type, target):
...
     return strategy
 
-@schedule_sparse_dense.register("cpu")
-def schedule_sparse_dense_cpu(attrs, outs, target):
-    """schedule sparse_dense for x86"""
-    with target:
-        return topi.x86.schedule_sparse_dense(outs)
+@sparse_dense_strategy.register("cpu")
+def sparse_dense_strategy_cpu(attrs, inputs, out_type, target):
+    """batch_matmul x86 strategy"""
```
Comment (vinx13) [inline python/tvm/relay/op/strategy/x86.py:299]:
```suggestion
    """sparse_dense x86 strategy"""
```

**Pull request merged**